diff --git a/src/main/java/com/fasterxml/jackson/core/filter/FilteringParserDelegate.java b/src/main/java/com/fasterxml/jackson/core/filter/FilteringParserDelegate.java
index e6558fd0..badab4ea 100644
--- a/src/main/java/com/fasterxml/jackson/core/filter/FilteringParserDelegate.java
+++ b/src/main/java/com/fasterxml/jackson/core/filter/FilteringParserDelegate.java
@@ -236,6 +236,7 @@ public class FilteringParserDelegate extends JsonParserDelegate
         if (!_allowMultipleMatches && (_currToken != null) && (_exposedContext == null)) {
             //if not scalar and ended successfully, and !includePath, then return null
                 if (_currToken.isStructEnd()) {
+		if (_currToken == JsonToken.START_OBJECT)
                     if (_headContext.isStartHandled()) {
                         return (_currToken = null);
                     }
