diff --git a/source/org/jfree/chart/block/BorderArrangement.java b/source/org/jfree/chart/block/BorderArrangement.java
index 730aeb3..a54706c 100644
--- a/source/org/jfree/chart/block/BorderArrangement.java
+++ b/source/org/jfree/chart/block/BorderArrangement.java
@@ -447,7 +447,10 @@ public class BorderArrangement implements Arrangement, Serializable {
                     LengthConstraintType.RANGE, h[2], null,
                     LengthConstraintType.FIXED);
             Size2D size = this.leftBlock.arrange(g2, c3);
-            w[2] = size.width;
+if (this.leftBlock != null) {
+			this.leftBlock.setBounds(new Rectangle2D.Double(0.0, h[0], w[2], h[2]));
+		}
+ 
         }
         h[3] = h[2];
         if (this.rightBlock != null) {
