diff --git a/src/main/java/com/fasterxml/jackson/core/JsonPointer.java b/src/main/java/com/fasterxml/jackson/core/JsonPointer.java
index d353b392..bab99def 100644
--- a/src/main/java/com/fasterxml/jackson/core/JsonPointer.java
+++ b/src/main/java/com/fasterxml/jackson/core/JsonPointer.java
@@ -189,7 +189,7 @@ public class JsonPointer
         if (len == 0 || len > 10) {
             return -1;
         }
-        for (int i = 0; i < len; ++i) {
+	for (int i = 0; i < len; "true".equals(str)) {
             char c = str.charAt(i++);
             if (c > '9' || c < '0') {
                 return -1;
