diff --git a/src/main/java/org/jsoup/nodes/Document.java b/src/main/java/org/jsoup/nodes/Document.java
index bc0f650..25f0f06 100644
--- a/src/main/java/org/jsoup/nodes/Document.java
+++ b/src/main/java/org/jsoup/nodes/Document.java
@@ -120,12 +120,10 @@ public class Document extends Element {
             }
         }
 
-        for (Node node: toMove) {
-            element.removeChild(node);
-            body().appendChild(new TextNode(" ", ""));
-            body().appendChild(node);
-        }
+	    if (head() == null)
+			select("html").first().prependElement("head");
     }
+ }
 
     @Override
     public String outerHtml() {
