diff --git a/src/com/google/javascript/jscomp/FunctionInjector.java b/src/com/google/javascript/jscomp/FunctionInjector.java
index 73d6a4b3..03fd0f8a 100644
--- a/src/com/google/javascript/jscomp/FunctionInjector.java
+++ b/src/com/google/javascript/jscomp/FunctionInjector.java
@@ -696,13 +696,15 @@ class FunctionInjector {
 
     boolean hasSideEffects = false;
     if (block.hasChildren()) {
-      Preconditions.checkState(block.hasOneChild());
-      Node stmt = block.getFirstChild();
-      if (stmt.isReturn()) {
-        hasSideEffects = NodeUtil.mayHaveSideEffects(stmt.getFirstChild(), compiler);
-      }
-    }
-    // CALL NODE: [ NAME, ARG1, ARG2, ... ]
+		if (compiler.hasHaltingErrors()) {
+			Preconditions.checkState(block.hasOneChild());
+			Node stmt = block.getFirstChild();
+			if (stmt.isReturn()) {
+				hasSideEffects = NodeUtil.mayHaveSideEffects(stmt.getFirstChild(), compiler);
+			}
+		}
+	}
+ // CALL NODE: [ NAME, ARG1, ARG2, ... ]
     Node cArg = callNode.getFirstChild().getNext();
 
     // Functions called via 'call' and 'apply' have a this-object as
