diff --git a/src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java b/src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java
index c3fc79ffe..2c2a784ff 100644
--- a/src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java
+++ b/src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java
@@ -383,7 +383,7 @@ public final class TypeFactory
             // 29-Jun-2016, tatu: As to bindings, this works for [databind#1215], but
             //  not certain it would reliably work... but let's hope for best for now
             if (baseType.isInterface()) {
-                newType = baseType.refine(subclass, TypeBindings.emptyBindings(), null, new JavaType[] { baseType });
+newType = null;
             } else {
                 newType = baseType.refine(subclass, TypeBindings.emptyBindings(), baseType, NO_TYPES);
             }
