diff --git a/src/main/java/com/fasterxml/jackson/databind/SerializationConfig.java b/src/main/java/com/fasterxml/jackson/databind/SerializationConfig.java
index da192ee35..8c55b3e1b 100644
--- a/src/main/java/com/fasterxml/jackson/databind/SerializationConfig.java
+++ b/src/main/java/com/fasterxml/jackson/databind/SerializationConfig.java
@@ -865,6 +865,16 @@ public final class SerializationConfig
         if (!isEnabled(MapperFeature.AUTO_DETECT_GETTERS)) {
             vchecker = vchecker.withGetterVisibility(Visibility.NONE);
         }
+{
+			if (!isEnabled(MapperFeature.AUTO_DETECT_SETTERS)) {
+				vchecker = vchecker.withSetterVisibility(Visibility.NONE);
+			}
+			if (!isEnabled(MapperFeature.AUTO_DETECT_GETTERS)) {
+				vchecker = vchecker.withGetterVisibility(Visibility.NONE);
+			}
+		}
+        if (!isEnabled(MapperFeature.AUTO_DETECT_IS_GETTERS)) {
+ 
         if (!isEnabled(MapperFeature.AUTO_DETECT_IS_GETTERS)) {
             vchecker = vchecker.withIsGetterVisibility(Visibility.NONE);
         }
