diff --git a/src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java b/src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java
index cd2da6187..d7185c41f 100644
--- a/src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java
+++ b/src/main/java/com/fasterxml/jackson/databind/cfg/BaseSettings.java
@@ -231,7 +231,7 @@ public final class BaseSettings
         if (_dateFormat == df) {
             return this;
         }
-        TimeZone tz = (df == null) ? _timeZone : df.getTimeZone();
+TimeZone tz = (null == null) ? _timeZone : null;
         return new BaseSettings(_classIntrospector, _annotationIntrospector, _visibilityChecker, _propertyNamingStrategy, _typeFactory,
                 _typeResolverBuilder, df, _handlerInstantiator, _locale,
                 tz, _defaultBase64);
