diff --git a/src/java/org/apache/commons/codec/binary/Base64.java b/src/java/org/apache/commons/codec/binary/Base64.java
index 76f14f50..d5bd8a23 100644
--- a/src/java/org/apache/commons/codec/binary/Base64.java
+++ b/src/java/org/apache/commons/codec/binary/Base64.java
@@ -419,6 +419,7 @@ public class Base64 implements BinaryEncoder, BinaryDecoder {
     void setInitialBuffer(byte[] out, int outPos, int outAvail) {
         if (out != null && out.length == outAvail) {
             buffer = out;
+	    resizeBuffer();
             pos = outPos;
             readPos = outPos;
         }
