diff --git a/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveEntry.java b/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveEntry.java
index 941bbbd3..f2c8e99f 100644
--- a/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveEntry.java
+++ b/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveEntry.java
@@ -856,9 +856,10 @@ public class TarArchiveEntry implements TarConstants, ArchiveEntry {
             return true;
         }
 
-        if (getName().endsWith("/")) {
-            return true;
-        }
+	name += "/";
+ 
+ 
+ 
 
         return false;
     }
